bittide/src/Bittide/ClockControl/Callisto.hs

```diff
@@ -95,40 +121,57 @@ callisto ::
   Signal dom (BitVector n) ->
   -- | Data counts from elastic buffers
   Signal dom (Vec n (DataCount m)) ->
-  -- | Speed change requested from clock multiplier
-  Signal dom SpeedChange
-callisto targetCount updateEveryNCycles mask allDataCounts =
-  mux shouldUpdate (D.toSignal b_kNext) (pure NoChange)
+  -- | Speed change requested from clock multiplier, the joint
+  -- stability indicator for all buffers, and the joint
+  -- "being-centered" indicator for all buffers
+  Signal dom (SpeedChange, AllStable, AllCentered)
+callisto margin framesize targetCount updateEveryNCycles mask allDataCounts =
+  bundle
+    ( mux shouldUpdate (D.toSignal b_kNext) (pure NoChange)
+    , allStable
+    , allCentered
+    )
  where
   dataCounts = filterCounts <$> fmap bv2v mask <*> allDataCounts
   filterCounts vMask vCounts = flip map (zip vMask vCounts) $
     \(isActive, count) -> if isActive == high then count else 0
+  scs = bundle $ map (stabilityChecker margin framesize) $ unbundle dataCounts
   updateCounter = wrappingCounter updateEveryNCycles
   shouldUpdate = updateCounter .==. 0
-  state = register initState (mux shouldUpdate updatedState state)
+  allStable = fold (&&) . (True :>) . map fst <$> scs
+  allCentered = fold (&&) . (True :>) . map snd <$> scs
```

Comment:
```suggestion
  allStable = and . map fst <$> scs
  allCentered = and . map snd <$> scs
```

Reply:
Also, a small record with two fields `stable` / `centered` would make this more readable / less error prone.

Reply:
Done.